pandas/core/indexes/numeric.py

```diff
@@ -43,15 +46,15 @@
 _num_index_shared_docs = {}
 
 
-class NumericIndex(Index):
+class BaseNumericIndex(Index):
```

Comment:
Is there a reason to still have a BaseNumericIndex? (if the Int64/Float64Index directly inherit from NumericIndex, and not from BaseNumericIndex)

Reply:
`RangeIndex` inherited from `BaseNumericIndex`, so I still kept it.

However, the tests in `pandas/core/indexes/range.py`  all actually pass if it inherits from `RangeIndex` inherits from `NumericIndex`, so I actually think we can avoid BaseNumericIndex. I'll change it if all tests pass.

Reply:
`BaseNumericIndex` has been removed in the newest version and we just have a `NumericIndex`, which inherits from `Index`. I'm marking this resolved.